lewis_emulators/jsco4180/interfaces/stream_interface.py

```diff
@@ -1,166 +1,185 @@
 
     def __init__(self):
 
         super(Jsco4180StreamInterface, self).__init__()
         # Commands that we expect via serial during normal operation
         self.commands = {
-            CmdBuilder(self.set_flowrate).float().escape(" flowrate set").build(),
+            CmdBuilder(self.set_flowrate).float().escape(" flowrate set").eos().build(),
+            CmdBuilder(self.get_flowrate_rbv).escape("flowrate load p").eos().build(),
             CmdBuilder(self.get_flowrate).escape("a_flow load p").eos().build(),
+
             CmdBuilder(self.get_pressure).escape("a_press1 load p").eos().build(),
             CmdBuilder(self.set_pressure_max).int().escape(" pmax set").build(),
             CmdBuilder(self.get_pressure_max).escape("a_pmax load p").eos().build(),
             CmdBuilder(self.set_pressure_min).int().escape(" pmin set").build(),
             CmdBuilder(self.get_pressure_min).escape("a_pmin load p").eos().build(),
-            CmdBuilder(self.set_pump_off_timer).float().escape(" offtimer set").build(),
-            CmdBuilder(self.set_pump_on_timer).float().escape(" ontimer set").build(),
-            CmdBuilder(self.set_valve_position).int().escape(" valve set").build(),
-            CmdBuilder(self.get_valve_position).escape("valve load p").eos().build(),
-            CmdBuilder(self.set_file_number).int().escape(" fileno set").build(),
-            CmdBuilder(self.set_file_open).int().escape(" openfile").build(),
-            CmdBuilder(self.set_file_close).escape("closefile").build(),
+
             CmdBuilder(self.get_program_runtime).escape("current_time load p").eos().build(),
+
             CmdBuilder(self.get_component_a).escape("compa load p").eos().build(),
             CmdBuilder(self.get_component_b).escape("compb load p").eos().build(),
             CmdBuilder(self.get_component_c).escape("compc load p").eos().build(),
             CmdBuilder(self.get_component_d).escape("compd load p").eos().build(),
-            CmdBuilder(self.set_composition).float().escape(" ").float().escape(" ").float().escape(" ").float()
-                                            .escape(" comp set").build(),
+            CmdBuilder(self.set_composition).float().escape(" ").float().escape(" ").float().escape(" ").float().escape(" comp set").eos().build(),
+
             CmdBuilder(self.get_error).escape("trouble load p").eos().build(),
             CmdBuilder(self.set_error).escape("0 trouble set").build(),
+
             CmdBuilder(self.set_pump).int().escape(" pump set").eos().build(),
+            CmdBuilder(self.get_status).escape("status load p").eos().build(),
+
+            CmdBuilder(self.set_file_number).int().escape(" fileno set").eos().build(),
+            CmdBuilder(self.set_file_open).int().escape(" openfile").eos().build(),
+            CmdBuilder(self.set_file_closed).int().escape(" closefile").eos().build(),
         }
 
     def catch_all(self):
         pass
 
+    @if_valid_input_delay
+    @if_connected
```

Comment:
If these two decorators are always used together we should combine them into one to save on code duplication

```
def combined_checks(func):
    return if_valid_input_delay(if_connected(func))

@combined_checks
def set_file_open(...):
    ...
```